### PR TITLE
Darkside reloaded 3

### PR DESCRIPTION
**Pull request by fluidvanadium**
Test upgrades:
  - feature-gated txid replacement logic
  - darkside bump chain now works with multiple transactions
  - with_assertions has more checks
  - specific values replaced with single_sufficient_send proptest
  - DRY and refactors

### CODE REVIEW

File: darkside-tests/src/chain_generics.rs [hunk 1]
```diff
@@ -82,41 +103,105 @@ pub(crate) mod conduct_chain {
...
                             .unwrap();
-                        self.tree_state = update_tree_states_for_transaction(
-                            &self.darkside_connector.0,
-                            raw_tx.clone(),
-                            u64::from(self.staged_blockheight),
+
+                        //trees
+                        let transaction = zcash_primitives::transaction::Transaction::read(
+                            raw_tx.data.as_slice(),
+                            zcash_primitives::consensus::BranchId::Nu5,
```
Comment (AloeareV) [inline darkside-tests/src/chain_generics.rs]: Hardcoded branch ID of nu5 concerns me, as we just hit nu6! I intend to turn this into a more meaningful comment before I submit the review, once I understand how this effects things.

File: darkside-tests/src/darkside_connector.rs [hunk 2]
```diff
@@ -51,7 +47,7 @@ impl DarksideConnector {
                     let uri = Uri::builder()
                         .scheme(uri.scheme().unwrap().clone())
                         .authority(uri.authority().unwrap().clone())
-                        //here. The Request's uri contains the path to the GRPC server and
+                        //here. The Request's uri contains the path to the GRPC sever and
```
Comment (AloeareV) [inline darkside-tests/src/darkside_connector.rs]: Revert typo!
```suggestion
                        //here. The Request's uri contains the path to the GRPC server and
```

File: libtonode-tests/tests/chain_generics.rs
Comment (AloeareV) [inline libtonode-tests/tests/chain_generics.rs:1]: I need to think through this set of proptest tests, to confirm that all those removed tests don't remove any cases not covered by the new proptest invocation, this comment is therefore a breadcrumb for myself! This is so much more succinct, big fan.

**Review by AloeareV — COMMENTED** [review]
This is great! I've left myself a couple comments of things I want to understand better before either approving them or requesting changes.

File: libtonode-tests/tests/chain_generics.rs
Comment (AloeareV) [reply libtonode-tests/tests/chain_generics.rs:1]: > I need to think through this set of proptest tests, to confirm that all those removed tests don't remove any cases not covered by the new proptest invocation, this comment is therefore a breadcrumb for myself! This is so much more succinct, big fan.

I have confirmed to my satisfaction that these two tests do the job of the roughly 35(!!) tests they replace!

File: darkside-tests/src/chain_generics.rs (hunk 1 above)
Comment (fluidvanadium) [reply darkside-tests/src/chain_generics.rs]: i think this wont effect the transaction, but it should be fixed
it would be good to make this a Const

Comment (AloeareV) [reply darkside-tests/src/chain_generics.rs]: Looks like this arg is only read in the pre-zip255 case!

**Review by AloeareV — APPROVED** [review]

**Pull request merged**